vchan/libvchan.h

```diff
@@ -47,6 +47,20 @@ libvchan_t *libvchan_server_init(int domain, int port, size_t read_min, size_t w
 
 libvchan_t *libvchan_client_init(int domain, int port);
 
+/* An alternative path for client connection:
+ * 1. Call libvchan_client_init_async().
+ * 2. Wait for watch_fd to become readable.
+ * 3. When readable, call libvchan_client_init_async_finish().
+ *
+ * Repeat steps 2-3 until libvchan_client_init_async_finish returns 0. Abort on
+ * negative values (error). In that case, libvchan_t struct is already released.
```

Comment:
This is very error-prone, as it means that the struct lifetime is not known at compile-time.

Reply:
So, do you suggest requiring the app to call `libvchan_client_init_async_cancel()` in that case?

Reply:
In fact, I can adjust `libvchan_close()` to deal with this case too, so the cleanup would be consistent in both connected and semi-connected state.